Fabric.Identity.API/Services/GroupFilterService.cs

```diff
@@ -23,23 +23,33 @@ public IEnumerable<Claim> FilterClaims(IEnumerable<Claim> claims)
                 return claims;
             }
 
-            var prefixes = _groupFilterSettings.Prefixes?.ToList();
-            var suffixes = _groupFilterSettings.Suffixes?.ToList();
+            var prefixFilters = _groupFilterSettings.Prefixes?.ToList();
+            var suffixFilters = _groupFilterSettings.Suffixes?.ToList();
+
+            var hasPrefixFilters = prefixFilters != null && prefixFilters.Count > 0;
+            var hasSuffixFilters = suffixFilters != null && suffixFilters.Count > 0;
+            var hasFilters = hasPrefixFilters || hasSuffixFilters;
+
+            if (!hasFilters)
+            {
+                return claims;
+            }
+
             var claimList = claims.ToList();
-            var prefixFilteredClaims = claimList;
-            var suffixFilteredClaims = claimList;
+            var prefixFilteredClaims = new List<Claim>();
+            var suffixFilteredClaims = new List<Claim>();
 
-            if (prefixes?.Count > 0)
+            if (prefixFilters?.Count > 0)
```

Comment:
Consider putting in some logging so we can quickly identify if these filters are working.  Perhaps log the claim count before and after applying the filters?